sgx/report-cache/untrusted/src/lib.rs

```diff
@@ -194,68 +155,15 @@ impl<E: ReportableEnclave, R: RaClient> ReportCache<E, R> {
             self.logger,
             "Starting enclave report cache update process..."
         );
-        let mut attestation_evidence = self.start_report_cache()?;
+        let attestation_evidence = self.start_report_cache()?;
 
         log::debug!(
             self.logger,
             "Verifying attestation evidence with enclave..."
         );
-        let retval = match self
+        Ok(self
             .enclave
-            .verify_attestation_evidence(attestation_evidence.clone())
-        {
-            Ok(()) => {
-                log::debug!(self.logger, "Enclave accepted report as valid...");
-                Ok(())
-            }
-            Err(
-                error @ ReportableEnclaveError::AttestEnclave(AttestEnclaveError::Verify(
-                    VerifierError::Verification(_),
-                )),
-            ) => {
-                let report_data = VerificationReportData::try_from(&attestation_evidence)?;
-                if let Some(platform_info_blob) = report_data.platform_info_blob.as_ref() {
-                    // IAS gave us a PIB
-                    log::debug!(
-                        self.logger,
-                        "IAS requested TCB update, attempting to update..."
-                    );
-                    QuotingEnclave::update_tcb(platform_info_blob)?;
-                    log::debug!(
-                        self.logger,
-                        "TCB update complete, restarting reporting process"
-                    );
-                    attestation_evidence = self.start_report_cache()?;
-                    log::debug!(
-                        self.logger,
-                        "Verifying attestation evidence with enclave (again)..."
-                    );
-                    self.enclave
-                        .verify_attestation_evidence(attestation_evidence.clone())?;
-                    log::debug!(self.logger, "Enclave accepted new report as valid...");
-                    Ok(())
-                } else {
-                    Err(error.into())
-                }
-            }
-            Err(error) => Err(error.into()),
-        };
-
-        if retval.is_ok() {
-            // TODO: replace with dcap
-            let ias_report_data = VerificationReportData::try_from(&attestation_evidence)?;
-            let timestamp = ias_report_data.parse_timestamp()?;
-
-            self.report_timestamp_gauge.set(timestamp.timestamp());
-
-            log::info!(
-                self.logger,
-                "Enclave accepted report as valid, report generated at {:?}...",
-                timestamp
-            );
-        }
-
-        retval
+            .verify_attestation_evidence(attestation_evidence)?)
     }
```

Comment:
The timestamp guage was near here. Since the new evidence is all local to the enclaves, and there is no trusted time in the enclaves,  there is no timestamp in the new evidence.

TBH I'm not sure what/where these guages were being used. It may be that w should bring the timestamp guage stuff down into the ReportCacheThread, where it kicks for the updates, and it puts a new value out anytime the update is called and succeeds

Reply:
Added back the gauge in https://github.com/mobilecoinfoundation/mobilecoin/pull/3612/commits/0b799aaa32008732b38dbbc82f258fce6c097aa7